File: src/2025-02-22-00-51-50--a-deepseek-r1-test-220-10-1--2-1-huoshan/go/exercises/practice/transpose/transpose.go
package transpose

func Transpose(input []string) []string {
	if len(input) == 0 {
		return []string{}
	}

	// Find maximum row length
	maxLen := 0
	for _, row := range input {
		if len(row) > maxLen {
			maxLen = len(row)
		}
	}

	var result []string

	for i := 0; i < maxLen; i++ {
		var transposedRow []rune
		for j, row := range input {
			if i < len(row) {
				transposedRow = append(transposedRow, rune(row[i]))
			} else {
				// Check if any row >=j has length >i
				hasLonger := false
				for k := j; k < len(input); k++ {
					if len(input[k]) > i {
						hasLonger = true
						break
					}
				}
				if hasLonger {
					transposedRow = append(transposedRow, ' ')
				}
			}
		}
		result = append(result, string(transposedRow))
	}

	return result
}

File: src/2025-03-04-12-20-21--deepseek-vr-0304/javascript/exercises/practice/transpose/transpose.js
//
// This is only a SKELETON file for the 'Transpose' exercise. It's been provided as a
// convenience to get you started writing code faster.
//

export const transpose = (input) => {
  if (input.length === 0) return [];
  const maxLen = Math.max(...input.map(row => row.length));
  const result = [];
  for (let c = 0; c < maxLen; c++) {
    let transposedRow = '';
    for (const row of input) {
      transposedRow += c < row.length ? row[c] : ' ';
    }
    transposedRow = transposedRow.replace(/\s+$/, '');
    result.push(transposedRow);
  }
  return result;
};
